src/ProductsContext/ProductState.tsx

```diff
@@ -56,14 +66,41 @@ const ProductState: React.FC<ProductStateProps> = (props: any) => {
       const res = await checkFn();
       if(res === null) return null 
       else {
-        const response = await fetch(`${host}/products/${id}`)
-        const json = await response.json();
+        if(id < 100) {const response = await fetch(`${host}/products/${id}`)}
+        // const json = await response.json();
+
+        let json = {}
+        for(let i = 0; i < data.length; i++) {
+          const pro = data[i]
+          if(pro.id == id) {
+            json = pro;
+            break;
+          }
+        }
         return json
       }
   }
 
+  const handleAddItem = async (title: string, desc: string) => {
+    const res = await checkFn();
+    if(res === null) return null 
+    else {
+      const thumbnail = thumbnailImg;
+      const images = [Img]
+      const newItem = {
+        id: data.length + 1,
+        title: title,
+        description: desc,
+        thumbnail: thumbnail,
+        images: images
+      }
 
-  const value: any = { getProducts, products, setName, setPassword, name, password, handleLogin, handleGetDetails };
+      data.push(newItem)
+      localStorage.setItem('products', JSON.stringify(data))
```

Comment:
Also add the call to dummy api add product API. do both for every action. once you get a positive response, update local storage and display from there.

Reply:
made changes, please give me final confirmation to merge